### PR TITLE
[eda] Quick Fit

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*

## `AutoGluonModelQuickFit`

The component allows to fit a quick model using AutoGluon. `train_data`, `val_data` and `label` must be present in args.
Note: this component can be wrapped into :`TrainValidationSplit` and `Sampler` to perform automated sampling and train-test split. This whole logic is implemented in `quick_fit()` shortcut (see below).

```
    Parameters
    ----------
    problem_type: str, default = 'auto'
        problem type to use. Valid problem_type values include ['auto', 'binary', 'multiclass', 'regression', 'quantile', 'softclass']
        auto means it will be Auto-detected using AutoGluon methods.
    estimator_args
        kwargs to pass into estimator constructor (`TabularPredictor`)
    parent: Optional[AbstractAnalysis], default = None
        parent Analysis
    children: Optional[List[AbstractAnalysis]], default None
        wrapped analyses; these will receive sampled `args` during `fit` call
```

Usage:
```python
auto.analyze(
    train_data=df_train,
    label=label,
    anlz_facets=[
        eda.dataset.TrainValidationSplit(children=[  # Creates train-test split using train_data above
            eda.model.AutoGluonModelQuickFit(  # Fits the model
                estimator_args=dict(path=path),
                verbosity=0,
                hyperparameters={
                    "RF": {
                        "criterion": "entropy",
                        "max_depth": 15,
                        "ag_args": {"name_suffix": "Entr", "problem_types": ["binary", "multiclass"]},
                    }
                },
                children=[
                    eda.model.AutoGluonModelEvaluator()  # Fitted model is evaluated
                ],
            )
        ])
    ],
)
```

## `auto.quick_fit()`
This helper performs quick model fit analysis and then produces a composite report of the results.

The analysis is structured in a sequence of operations:
- Sample if `sample` is specified.
- Perform train-test split using `val_size` ratio
- Fit AutoGluon estimator given `fit_args`; if `hyperparameters` not present in args, then use default ones
    (Random Forest by default - because it is interpretable)
- Display report

The reports include:
- confusion matrix for classification problems; predictions vs actual for regression problems
- model leaderboard
- feature importance

```
    Parameters
    ----------
    train_data: DataFrame
        training dataset
    label: str
        target variable
    path: Optional[str], default = None,
        path for models saving
    problem_type: str, default = 'auto'
        problem type to use. Valid problem_type values include ['auto', 'binary', 'multiclass', 'regression', 'quantile', 'softclass']
        auto means it will be Auto-detected using AutoGluon methods.
    sample: Union[None, int, float], default = None
        sample size; if `int`, then row number is used;
        `float` must be between 0.0 and 1.0 and represents fraction of dataset to sample;
        `None` means no sampling
        See also :func:`autogluon.eda.analysis.dataset.Sampler`
    val_size: float, default = 0.3
        fraction of training set to be assigned as validation set during the split.
    state: Union[None, dict, AnalysisState], default = None
        pass prior state if necessary; the object will be updated during `anlz_facets` `fit` call.
    return_state: bool, default = False
        return state if `True`
    verbosity: int, default = 0
        Verbosity levels range from 0 to 4 and control how much information is printed.
        Higher levels correspond to more detailed print statements (you can set verbosity = 0 to suppress warnings).
        If using logging, you can alternatively control amount of information printed via `logger.setLevel(L)`,
        where `L` ranges from 0 to 50 (Note: higher values of `L` correspond to fewer print statements, opposite of verbosity levels).
    show_feature_importance_barplots: bool, default = False
        if `True`, then barplot char will ba added with feature importance visualization
    fit_args
        kwargs to pass into `TabularPredictor` fit

    Returns
    -------
        state after `fit` call if `return_state` is `True`; `None` otherwise
```

**Usage:**
```python
import autogluon.eda.auto as auto

auto.quick_fit(train_data=df_train, label=label, show_feature_importance_barplots=True, hyperparameters='toy')
```

image

## `auto.dataset_overview()`

Shortcut to perform high-level datasets summary overview (counts, frequencies, missing statistics, types info).

```
Parameters
----------
train_data: Optional[DataFrame], default = None
    training dataset
test_data: Optional[DataFrame], default = None
    test dataset
val_data: Optional[DataFrame], default = None
    validation dataset
label: : Optional[str], default = None
    target variable
state: Union[None, dict, AnalysisState], default = None
    pass prior state if necessary; the object will be updated during `anlz_facets` `fit` call.
sample: Union[None, int, float], default = None
    sample size; if `int`, then row number is used;
    `float` must be between 0.0 and 1.0 and represents fraction of dataset to sample;
    `None` means no sampling
    See also :func:`autogluon.eda.analysis.dataset.Sampler`
```

**Usage:**
```python
auto.dataset_overview(train_data=df_train, test_data=df_test, label=target_col)
```

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by tonyhoo — APPROVED** [review]

File: eda/src/autogluon/eda/auto/simple.py [hunk 1]
```diff
@@ -163,3 +172,133 @@ def analyze_interaction(
...
+
+    if not isinstance(state, AnalysisState):
+        state = AnalysisState(state)
+
+    if "hyperparameters" not in fit_args:
+        fit_args = fit_args.copy()
+        fit_args["hyperparameters"] = {
+            "RF": [
+                {
+                    "criterion": "entropy",
```
[the remark this answers is not in the document]
Comment (tonyhoo) [reply eda/src/autogluon/eda/auto/simple.py:260]: LGTM
